### PR TITLE
New "modified documents" dialog provides an easy way to see all committable documents, use batch commit or batch revert to live, etc.

### PR DESCRIPTION
**Pull request by boutell**
We now track whether each document is modified via a workflowModified property which is updated on afterSave.

This allows simple queries for modified draft documents.

Also greatly improved performance for making the calculation of which documents are modified and numerous fixes to make the calculation of which documents are truly modified more accurate. 

Requires related PR to apostrophe which is forthcoming in a moment.

### CODE REVIEW

**Comment by boutell** [comment]
I lied: there is no strictly required update to apostrophe core to use this, although in practice we'll want what is currently in master to quiet the "oh no, you didn't consume a promise" warning messages. If you decide to test drive this, just make sure you npm link apostrophe with latest master.

Also if test-driving make sure you run `apostrophe-migrations:migrate`.

File: lib/modules/apostrophe-workflow-committables/views/manageListPage.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{%- import "apostrophe-ui:components/fields.html" as fields -%}
+{% for piece in data.pieces %}
+  <tr data-piece="{{ piece._id }}">
+    <td>{{fields.checkbox(data.options.name + '-select')}}</td>
+    {% if apos.utils.beginsWith(piece.slug, '/') %}
+      <td><a href="{{ piece._url }}">{{ piece.title or piece.slug }}</a></td>
+    {% else %}
+      <td><a href="#" data-apos-edit-{{ piece.type }}="{{ piece._id }}">{{ piece.title or piece.slug }}</a></td>
```
Comment (abea) [inline lib/modules/apostrophe-workflow-committables/views/manageListPage.html]: Would be improved by switching to `button` (a11y and whatnot)

File: lib/modules/apostrophe-workflow-docs/index.js [hunk 2]
```diff
@@ -48,5 +50,31 @@ module.exports = {
       return superTestInsertPermissions(req, doc, options);
     };
 
+    // An afterSave handler is a good place to set or clear the
+    // workflowModified flag because it guarantees any properties
+    // added by beforeSave handlers are taken into account. It would
+    // be nice if promise events had a way to say "after all the
+    // others," but they don't so far.
+
+    self.on('apostrophe-docs:afterSave', 'setWorkflowModified', function(req, doc, options) {
```
Comment (abea) [inline lib/modules/apostrophe-workflow-docs/index.js]: This function should only look at drafts

File: public/js/committable-modal.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// A modal for reviewing workflow submissions.
+
+apos.define('apostrophe-workflow-committable-modal', {
```
Comment (abea) [inline public/js/committable-modal.js]: Redundant now that it's a virtual piece type.

File: public/js/user.js [hunk 4]
```diff
@@ -13,7 +13,9 @@ apos.define('apostrophe-workflow', {
     self.enableExport();
     self.enableReview();
     self.enableRevert();
+    self.enableRevertToLive();
     self.enableManageModal();
+    self.enableCommittableModal();
```
Comment (abea) [inline public/js/user.js]: Remove pls

File: public/js/user.js [hunk 5]
```diff
@@ -557,6 +590,16 @@ apos.define('apostrophe-workflow', {
       return apos.create(self.__meta.name + '-manage-modal', _.assign({ manager: self }, options));
     };
 
+    self.enableCommittableModal = function() {
```
Comment (abea) [inline public/js/user.js]: Please remove

File: public/js/user.js [hunk 6]
```diff
@@ -557,6 +590,16 @@ apos.define('apostrophe-workflow', {
...
+      apos.adminBar.link(self.__meta.name + '-committable-modal', function() {
+        self.launchCommittableModal();
+      });
+    };
+
+    self.launchCommittableModal = function() {
```
Comment (abea) [inline public/js/user.js]: Also remove

File: views/committable-list.html [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{% for doc in data.committable %}
```
Comment (abea) [inline views/committable-list.html]: This can go away.

File: views/committable-modal.html [hunk 8]
```diff
@@ -0,0 +1,53 @@
+{%- extends "apostrophe-modal:base.html" -%}
```
Comment (abea) [inline views/committable-modal.html]: This can go away.

File: views/committable-pager.html [hunk 9]
```diff
@@ -0,0 +1,2 @@
+{% import 'apostrophe-pager:macros.html' as pager with context %}
```
Comment (abea) [inline views/committable-pager.html]: This can go away

**Pull request merged**